test/jasmine/assets/custom_assertions.js

```diff
@@ -59,6 +59,92 @@ exports.assertHoverLabelStyle = function(g, expectation, msg, textSelector) {
     expect(textStyle.fill).toBe(expectation.fontColor, msg + ': font.color');
 };
 
+function assertLabelContent(label, expectation, msg) {
+    var lines = label.selectAll('tspan');
+    var lineCnt = lines.size();
+    var expectMultiLine = Array.isArray(expectation);
+
+    function extract(sel) {
+        return sel.node() ? sel.html() : null;
+    }
+
+    if(lineCnt > 0) {
+        if(expectMultiLine) {
+            expect(lines.size()).toBe(expectation.length, msg + ': # of lines');
+            lines.each(function(_, i) {
+                var l = d3.select(this);
+                expect(extract(l)).toBe(expectation[i], msg + ': tspan line ' + i);
+            });
+        } else {
+            fail('Expected a single-line label, found multiple lines');
+        }
+    } else {
+        if(!expectMultiLine) {
+            expect(extract(label)).toBe(expectation, msg + ': text content');
+        } else {
+            fail('Expected a multi-line label, found single');
+        }
+    }
+}
+
+function count(selector) {
+    return d3.selectAll(selector).size();
+}
+
+exports.assertHoverLabelContent = function(expectation, msg) {
```

Comment:
I like this fn - really nice idea! I find the signature a little unintuitive though... `expectation` has to be something like `[['a = 0.200', 'b = 3.500', 'y = 2.900'], 'a = 0.2']`... perhaps it would be clearer like `{nums: 'a = 0.200\nb = 3.500\ny = 2.900', name: 'a = 0.2'}`? Array vs multiline string I don't feel that strongly about, but I think named items instead of the outer array would be quite a bit easier to read, and possibly easier to write too.

Reply:
I get annoyed of typing object keys in assertion functions arguments when writing tests.

But I guess there's value in being a little more verbose for project-wide custom assertions.

Reply:
... and good call about the somewhat useless single-vs-multi line logic. Our `convertToTspans` tests are there for that.

Reply:
done in 294714b

Thoughts?